pages/events/[id].tsx

```diff
@@ -0,0 +1,89 @@
+import { useRouter } from 'next/router';
+import Image from 'next/image';
+import { useGetEventByIdQuery } from 'generated/graphql';
+import { Typography } from 'components/atoms/Typography/Typography';
+import { useState } from 'react';
+
+import { EventDetails } from 'components/templates/events/eventDetails';
+import { BudgetDetails } from 'components/templates/events/budgetDetails';
+
+const EventDetailsPage = () => {
+	const router = useRouter();
+	const id = router.query.id as string;
+	const { data, loading } = useGetEventByIdQuery({ variables: { id } });
+	const [isEvent, setIsEvent] = useState('event');
+
+	if (loading) {
+		return (
+			<div>
+				<Typography component="h1">Szczegóły wydarzenia</Typography>
+				<div className="flex justify-center">
+					<Image
+						src={'/RingSpinner.svg'}
+						width={80}
+						height={80}
+						alt="Spinner"
+					/>
+				</div>
+			</div>
+		);
+	}
+	if (!data) {
+		return (
+			<div>
+				<Typography component="h1">Data not found</Typography>
+			</div>
+		);
+	}
+	return (
+		<div>
+			<div>
+				<Typography component="h1">Wydarzenie</Typography>
+			</div>
+			<div>
+				<Image
+					className="h-32 w-full rounded-lg object-cover lg:h-48"
+					src={data.event?.image?.url || '/part.png'}
+					height={600}
+					width={600}
+					alt=""
+				/>
+				<div className="mt-6 sm:mt-2 2xl:mt-5">
+					<div className="border-b border-gray-200">
+						<div className="mx-auto max-w-5xl px-4 sm:px-6 lg:px-8">
+							<nav className="-mb-px flex space-x-8">
+								<button
+									onClick={() => setIsEvent('event')}
+									className={
+										isEvent === 'event'
+											? 'border-b-2 border-blue-500 text-gray-900'
+											: 'whitespace-nowrap border-b-2 border-transparent py-4 px-1 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700'
+									}
+								>
+									Wydarzenie
+								</button>
+								<button
+									onClick={() => setIsEvent('budget')}
+									className={
+										isEvent === 'event'
+											? 'whitespace-nowrap border-b-2 border-transparent py-4 px-1 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700'
+											: 'border-b-2 border-blue-500 text-gray-900'
+									}
+								>
+									Budzet
+								</button>
+							</nav>
+						</div>
+					</div>
+				</div>
+
+				{isEvent === 'event' ? (
+					<EventDetails data={data} />
+				) : (
+					<BudgetDetails data={data} />
+				)}
```

Comment:
Both of these views get the same props? It seems a little bit unnecessary. Let's do something like this
`<EventDetails ticketPrice={data.ticketPrice} ...and so on />`

Reply:
It's fixed now